receiver/windowsservicereceiver/factory_others.go

```diff
@@ -0,0 +1,48 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+//go:build !windows
+
+package windowsservicereceiver
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/scraper/scraperhelper"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/windowsservicereceiver/internal/metadata"
+)
+
+var errUnsupportedOS = errors.New("windowsservicereceiver: supported only on Windows")
+
+func NewFactory() receiver.Factory {
```

Comment:
At first I think this will be better if you have the same `NewFactory` at `factory.go` but then changing line 26 to `receiver.WithMetrics(createMetricsReceiver, metadata.MetricsStability)` and then implement the platform dependent `createMetricsReceiver` in `factory_windows.go` and `factory_others.go`. See `receiver/windowsperfcountersreceiver` for an example.